common/lib/api-client/transformers/move.transformer.js

```diff
@@ -0,0 +1,11 @@
+module.exports = function moveTransformer(data) {
+  const isYouthMove = [
+    'secure_training_centre',
+    'secure_childrens_home',
+  ].includes(data.from_location?.location_type)
+
+  return {
+    ...data,
+    _is_youth_move: isYouthMove,
```

Comment:
I've been between minds about this pattern. I used it on the [assessment transformer](https://github.com/ministryofjustice/hmpps-book-secure-move-frontend/blob/3516af7a7acdf8020b9d4327b71de88edfd794a9/common/lib/api-client/transformers/assessment.transformer.js#L15) to in order to be able to distinguish between properties added within our service and properties that have come from the API resource.

This isn't something we do in the [person transformer](https://github.com/ministryofjustice/hmpps-book-secure-move-frontend/blob/3516af7a7acdf8020b9d4327b71de88edfd794a9/common/lib/api-client/transformers/person.transformer.js#L4-L5) (yet) though because when moving that there I didn't want to apply that everywhere.

The other reason for the `_` prefix is to avoid name clashes. The person escort record actually already has a `framework` property which I didn't want to override.

What are other's thoughts on this?

Reply:
Seems like a good idea.

Are we definitely not using `_` to start properties elsewhere? (Could we use `$` or a more explicit prefix `auto_`, `virtual_`?

Should we use this pattern for all created properties... eg. `person.fullname`? (I guess that's what you're getting at)

And obviously, it should be properly documented…

Reply:
>Are we definitely not using _ to start properties elsewhere? (Could we use $ or a more explicit prefix auto_, virtual_?

That's a good question. I'm not aware of any API attributes using an underscore prefix. I went with underscore initially as I know it's something express does with the request object. 

>Should we use this pattern for all created properties... eg. person.fullname? (I guess that's what you're getting at)

Any yep, if we think it's a pattern worth using we should do it in all places and document it (as you pointed out).

>And obviously, it should be properly documented…

Agreed!